### PR TITLE
Update customer and login model

### PR DESCRIPTION
**Pull request by tanay1307shah**

### CODE REVIEW

File: models/customer.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    })
+    return id
+
+
+def getCustomerSchema(api):
+    reservation = api.model('Reservation', {
+        'id_store' : fields.String('ID of the store.'),   
+        'date': fields.Date(required=True, description='The availability date'),
+        'time': fields.DateTime(required=True, description='The availability time')
+    })
```
Comment (binoyskumar92) [inline models/customer.py]: You shouldn't create the reservation schema here. You can import it from the models.reservation.py

File: models/customer.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+from flask_restplus import fields
+def createCustomerSchema(api):
+
+    reservation = api.model('Reservation', {
+        'id_store' : fields.String('ID of the store.'),   
+        'date': fields.Date(required=True, description='The availability date'),
+        'time': fields.DateTime(required=True, description='The availability time')
+    })
+
```
Comment (binoyskumar92) [inline models/customer.py]: You shouldn't create the reservation schema here. You can import it from the models.reservation.py

What if i need to change schema for reservation. I have to modify all your changes here in customer.py as well
It is better to have one point of definition and using that copy across your modules/files.

File: helpers/customerLogic.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        for document in cursor: # iterate through each db result and append to a list
+            response.append(document)
+        if response != []:
+            return response[-1]["id"]
+        else:
+            return 0
+
+    except Exception as e:
+        print("Error occured:", str(e.args))
+        return sys.maxsize
```
Comment (binoyskumar92) [inline helpers/customerLogic.py]: I guess the idea of this file/method is to check whats the last inserted id and use it to compare and increment the id generator variable. But we don't need this with mongodb as ObjectID is generated by default.

File: app.py [hunk 4]
```diff
@@ -65,5 +75,76 @@ def delete(self):
...
+
+    @ns_api_v1.expect(createCustomerSchema(api))
+    def post(self):
+        try:
+            customerId = getLastCustomerId(customerTable)
+            
+            if customerId == sys.maxsize:
+                raise Exception()
+            else:
+                customerId += 1
```
Comment (binoyskumar92) [inline app.py]: This logic needs to be changed. So in mongo db a  unique ObjectID is created whenever you create a new document and hence you need not bother to add a logic to create unique ids.
Secondly this current logic of creating id is very dangerous in db security perspective. This leads to predicting ids of future users very easily. I know the scope of project is small and we can do this incrementing natural number until you reach the highest value but mongodb is already creating an id for you. You can see the oid created for the user document below

{
    "_id": {
        "$oid": "5eec747fa9d5841a633f97c9"
    },
    "id": 1,
    "name": "tanay shah",
    "phone": "6304632809",
    "username": "tanshah",
    "password": "$pbkdf2-sha256$29000$Xwsh5DyH0DpnrDXmnNM6xw$3Cp4yrMD8rO9xd9vJKcp7Q9xMz9a/yi6e/LyF45djZg"
}

File: app.py [hunk 5]
```diff
@@ -17,24 +22,29 @@ def specs_url(self):
 api = Api(app)
 
 table = setupDB("covid-ticketing-db", "store")
+customerTable = setupDB("covid-ticketing-db","customer")
+customerId = 0
```
Comment (binoyskumar92) [inline app.py]: Remove this as part of below comments

**Review by binoyskumar92 — COMMENTED** [review]
Please go through the comments and let me know if you have any questions

File: app.py [hunk 6]
```diff
@@ -65,5 +75,76 @@ def delete(self):
...
+    global customerTable
+   # @marshal_with(store_marshal)
+
+    def get(self):
+        try:
+            cursor = customerTable.find({}, {"password" : 0})   # operation on table to get all data
+            response = []
+            for document in cursor: # iterate through each db result and append to a list
+                response.append(document)
+                return Response('{"response":%s,"message":"Succesfully retreived all documents"}' % dumps(response), status=200, mimetype='application/json')
```
Comment (binoyskumar92) [inline app.py]: This response will output all the data in customer table right which includes hashed passwords as well. I would say you need to marshal this response before sending

https://flask-restplus.readthedocs.io/en/stable/marshalling.html

**Review by binoyskumar92 — COMMENTED** [review]
1 more comment

File: app.py (hunk 6 above)
Comment (tanay1307shah) [reply app.py]: it won't the query i used, will skip the password form the db directly. I mean I can marshall the response but this serves the same purpose.

**Review by binoyskumar92 — APPROVED** [review]
If the code is retested and working for all the other apis as well go ahead and merge

File: app.py (hunk 6 above)
Comment (binoyskumar92) [reply app.py]: Ok then that's good enough

**Pull request merged**